services/ftp/ftp_service.py

```diff
@@ -0,0 +1,173 @@
+
+    # On the next section we override actual handlers instead of callbacks for more alerts
+    def ftp_LIST(self, path):
+        self.__send_alert(USER_LISTED_DIR_DESCRIPTION, {
+            ADDITIONAL_FIELDS: self.__format_file_path(path)
+        })
+        FTPHandler.ftp_LIST(self, path)
+    
+    def ftp_NLST(self, path):
+        self.__send_alert(USER_LISTED_DIR_DESCRIPTION, {
+            ADDITIONAL_FIELDS: self.__format_file_path(path)
+        })
+        FTPHandler.ftp_NLST(self, path)
+    
+    def ftp_MLST(self, path):
+        self.__send_alert(USER_LISTED_DIR_DESCRIPTION, {
+            ADDITIONAL_FIELDS: self.__format_file_path(path)
+        })
+        FTPHandler.ftp_MLST(self, path)
+    
+    def ftp_CWD(self, path):
+        self.__send_alert(USER_NAVIGATED_DIR_DESCRIPTION, {
+            ADDITIONAL_FIELDS: self.__format_file_path(path)
+        })
+        FTPHandler.ftp_CWD(self, path)
+    
+    def ftp_MKD(self, path):
+        self.__send_alert(USER_CREATED_DIR_DESCRIPTION, {
+            ADDITIONAL_FIELDS: self.__format_file_path(path)
+        })
+        FTPHandler.ftp_MKD(self, path)
+    
+    def ftp_RMD(self, path):
+        self.__send_alert(USER_DELETED_DIR_DESCRIPTION, {
+            ADDITIONAL_FIELDS: self.__format_file_path(path)
+        })
+        FTPHandler.ftp_RMD(self, path)
+    
+    def ftp_DELE(self, path):
+        self.__send_alert(USER_DELETED_FILE_DESCRIPTION, {
+            ADDITIONAL_FIELDS: self.__format_file_path(path)
+        })
+        FTPHandler.ftp_DELE(self, path)
+
+
+class FTPAlertingServer(FTPServer):
+    def __init__(self, *args, **kwargs):
+        self.alerting_function = kwargs.pop("alerting_function")
+        self.base_dir = kwargs.pop("base_dir")
+        FTPServer.__init__(self, *args, **kwargs)
+
+
+class FTPService(ServerCustomService):
+    def __init__(self, *args, **kwargs):
+        super(FTPService, self).__init__(*args, **kwargs)
+        self.server = None
+        self.temp_dir = None
+
+    def prepare_temp_dir(self):
+        self.temp_dir = tempfile.mkdtemp()
+
+    def delete_temp_dir(self):
+        shutil.rmtree(self.temp_dir, ignore_errors=True)
+
+    def on_server_start(self):
+        self.prepare_temp_dir()
+        authorizer = DummyAuthorizer()
+        authorizer.add_user(DEFAULT_USER, DEFAULT_PASSWORD, homedir=self.temp_dir, perm='elradfmw')  # All permissions
+        authorizer.add_anonymous(homedir=self.temp_dir, perm='elradfmw')
+
+        handler = AlertingHandler
+        handler.authorizer = authorizer
+        self.server = FTPAlertingServer(
+            (SERVER_IP, 21),
+            handler,
+            alerting_function=self.add_alert_to_queue,
+            base_dir=self.temp_dir)
+        self.signal_ready()
+        self.server.serve_forever()
+
+    def on_server_shutdown(self):
+        if self.server:
+            self.server.close_all()
+        self.delete_temp_dir()
+
+    def __str__(self):
+        return "FTP"
+
```

Comment:
Add def test(self).
Make it connect to your server and perform the actions that lead to an alert (and then disconnect).
Return a list containing all the alert types raised by this function.